
## 01. Align reads to the reference genomes
### Build indexes reference
#### Red abalone
```bash
sbatch hisat2_build.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna
```
#### Green abalone
```bash
sbatch hisat2_build.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta
```

### Processed libs
```bash
# Green
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CFF* .
# Red
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CRR* .

```

### Aling individually paired-end reads to references
```bash
# Green abalone

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta $i; done

```

And
```bash
# Red abalone
for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna $i; done

for i in $(ls *R1_p.fq.gz);do unlink $i; done

```
And 
```bash
# Hybrid
cat ../Green/SoftmaskedFilteredHalful_medaka.FINAL.fasta ../Red/GCA_023055435.1_xgHalRufe1.0.p_genomic.fna > RF_Ref.fna

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIM
MED_DATA/READS_gz/PAIRED/CRF* .


```