
# Reference-guide transcriptome assembly
## Align reads to the reference genomes
### Build indexes reference
#### Red abalone
```bash
sbatch hisat2_build.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna
```
#### Green abalone
```bash
sbatch hisat2_build.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta
```

### Input (Processed) fastq dataset
```bash
# Green
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CFF* .
# Red
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CRR* .

```

### Aling individually paired-end reads to references
```bash
# Green abalone

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta $i; done

```

And
```bash
# Red abalone
for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna $i; done

for i in $(ls *R1_p.fq.gz);do unlink $i; done

```
And 
```bash
# Hybrid
cat ../Green/SoftmaskedFilteredHalful_medaka.FINAL.fasta ../Red/GCA_023055435.1_xgHalRufe1.0.p_genomic.fna > RF_Ref.fna

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIM
MED_DATA/READS_gz/PAIRED/CRF* .

sbatch hisat2_build.sh RF_Ref.fna

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh RF_Ref $i; done

srun hisat2  --phred33 --dta -p 20 -x $idx_file -1 ${base}_R1_p.fq.gz -2 ${base}_R2_p.fq.gz --rg-id=${base} --rg SM:${base} -S ${base}.sam --summary-file ${base}.summary.txt --met-file ${base}.met.txt &


```


# Denovo transcriptome assembly

## Compare w/ Trinity-method
```bash
#!/bin/bash
#SBATCH --job-name=Trinity
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20

module load trinityrnaseq-v2.15.1

Trinity --seqType fq --max_memory 100G --left CFF1_R1_p.fq.gz,CFF2_R1_p.fq.gz,CFF3_R1_p.fq.gz,CRF2_R1_p.fq.gz,CRF3_R1_p.fq.gz,CRR1_R1_p.fq.gz,CRR2_R1_p.fq.gz,CRR3_R1_p.fq.gz --right CFF1_R2_p.fq.gz,CFF2_R2_p.fq.gz,CFF3_R2_p.fq.gz,CRF2_R2_p.fq.gz,CRF3_R2_p.fq.gz,CRR1_R2_p.fq.gz,CRR2_R2_p.fq.gz,CRR3_R2_p.fq.gz --CPU 20

exit
```

## Compare w/ rnaspads-method
```bash
#!/bin/bash
#SBATCH --job-name=spadesHaliotis
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20

EXPORT=/LUSTRE/apps/bioinformatica/SPAdes-3.15.5-Linux/bin/
export PATH=$PATH:$EXPORT

CPU=$SLURM_NPROCS
MEM=$SLURM_MEM_PER_NODE
DIR=$SLURM_SUBMIT_DIR

# rnaspades.py --pe1-1 S1.F.fq.gz --pe1-2 S1.R.fq.gz --pe1-1 S2.F.fq.gz --pe1-2 S2.R.fq.gz --pe1-1 S3.F.fq.gz --pe1-2 S3.R.fq.gz --pe1-1 ... --pe1-2 ... --pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/rnaspades

rnaspades.py \
  --pe1-1 CFF1_R1_p.fq.gz --pe1-2 CFF1_R2_p.fq.gz \
  --pe1-1 CFF2_R1_p.fq.gz --pe1-2 CFF2_R2_p.fq.gz \
  --pe1-1 CFF3_R1_p.fq.gz --pe1-2 CFF3_R2_p.fq.gz \
  --pe1-1 CRF2_R1_p.fq.gz --pe1-2 CRF2_R2_p.fq.gz \
  --pe1-1 CRF3_R1_p.fq.gz --pe1-2 CRF3_R2_p.fq.gz \
  --pe1-1 CRR1_R1_p.fq.gz --pe1-2 CRR1_R2_p.fq.gz \
  --pe1-1 CRR2_R1_p.fq.gz --pe1-2 CRR2_R2_p.fq.gz \
  --pe1-1 CRR3_R1_p.fq.gz --pe1-2 CRR3_R2_p.fq.gz \
  --pe1-fr -t $THREADS -m $MEM --cov-cutoff auto -o $DIR/rnaspades

exit

```
### Run rnaspades individually
The intension to assembly separately is: (A) it would be quicker ;) and (2) Capture sample-specific isoforms.

```bash
ls *fq.gz > FastqFiles.txt
FastqFiles=FastqFiles.txt
N=( $(cat $FastqFiles) )
NSZ=${#N[@]}
for((i=2; i<NSZ; i+=2)); do
    R1=${N[$i]}
    R2=${N[$(($i+1))]}
    ON=${N[$i]%%_*}
    echo $R1
    spades.py -t 1 -o $ON --pe1-1 $R1 --pe1-2 $R2
done
```
