
## 01. Align reads to the reference genomes
### Build indexes reference
#### Red abalone

```bash
genome=GCA_023055435.1_xgHalRufe1.0.p_genomic.fna
g_name=`basename ${genome%.fna}`

hisat2-build -p 20 $genome $g_name &> hisat2_build.log &

```
#### Green abalone
```bash
genome=SoftmaskedFilteredHalful_medaka.FINAL.fasta
g_name=`basename ${genome%.fasta}`

srun hisat2-build -p 20 $genome $g_name &> hisat2_build.log &

```


```bash
# Red
awk '{print $3}' *.gtf| sort | uniq -c
 591736 CDS
 665954 exon
  77553 five_prime_utr
  42838 gene
     22 Selenocysteine
  55645 start_codon
  55603 stop_codon
  54432 three_prime_utr
  70020 transcript

# Green
  awk '{print $3}' *.gff3 | sort | uniq -c
 324879 CDS
 329100 exon
  42489 gene
  45161 mRNA
     96 ncRNA
     95 ncRNA_gene

```
### Aling individually paired-end reads to references
```bash
# Run on paired-end reads

#Ex: 
# hisat2 --phred33 -p 8 -x ./GENOME_INDEX/$g_name -1 reads_f.fq -2 reads_r.fq -S output.sam

mkdir -p HISAT2_SAM_BAM_FILES

for file in $(ls *_1.P.qtrim.fq.gz | grep gz)
do
withpath="${file}"
filename=${withpath##*/}
base="${filename%*_*.P.qtrim.fq.gz}"
hisat2 --phred33 -p 8 -x $idx_file  \
    --rna-strandness RF \
    -1 ${base}_1.P.qtrim.fq.gz -2 ${base}_2.P.qtrim.fq.gz \
    --rg-id=${base} --rg SM:${base} -S HISAT2_SAM_BAM_FILES/${base}.sam
done 2> hisat2.log &

for i in $(ls *P.qtrim.fq.gz);do unlink $i; done


```



#########################################
## Rutas a las secuencias para trabajar ##
##########################################

# READS_gz
/LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz

# Reads concatenados (tres cruzas de abulon)
/LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/CONCATENATE_READS