
## 01. Align reads to the reference genomes
### Build indexes reference
#### Red abalone
```bash
sbatch hisat2_build.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna
```
#### Green abalone
```bash
sbatch hisat2_build.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta
```

### Processed libs
```bash
# Green
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CFF* .
# Red
ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIMMED_DATA/READS_gz/PAIRED/CRR* .

```

### Aling individually paired-end reads to references
```bash
# Green abalone

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh SoftmaskedFilteredHalful_medaka.FINAL.fasta $i; done

```

And
```bash
# Red abalone
for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh GCA_023055435.1_xgHalRufe1.0.p_genomic.fna $i; done

for i in $(ls *R1_p.fq.gz);do unlink $i; done

```
And 
```bash
# Hybrid
cat ../Green/SoftmaskedFilteredHalful_medaka.FINAL.fasta ../Red/GCA_023055435.1_xgHalRufe1.0.p_genomic.fna > RF_Ref.fna

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/Oyervides/data/OA_abulon/RNASEQ/TRIM
MED_DATA/READS_gz/PAIRED/CRF* .

sbatch hisat2_build.sh RF_Ref.fna

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh RF_Ref $i; done

srun hisat2  --phred33 --dta -p 20 -x $idx_file -1 ${base}_R1_p.fq.gz -2 ${base}_R2_p.fq.gz --rg-id=${base} --rg SM:${base} -S ${base}.sam --summary-file ${base}.summary.txt --met-file ${base}.met.txt &


```

### Report
```bash
mkdir Report
cp ./*/*.summary.txt Report/
multiqc Report/*
```

### Test if contaminant
```bash
wget https://genome-idx.s3.amazonaws.com/hisat/grch38_genome.tar.gz

tar -xvf grch38_genome.tar.gz

for i in $(ls *R1_p.fq.gz); do sbatch hisat_align.sh grch38/genome $i; done



# screen
#wget https://github.com/StevenWingett/FastQ-Screen/archive/refs/heads/master.zip
#unzip master.zip 


fastq_screen --aligner bowtie2 --conf fastqscreen.conf --subset 0 --filter 3---- --tag --force --outdir ./fastq_screen/ 012-X04-P68-COI-ICT_S46_L001_R1_001.fastq.fastq
  
# Consider mapping to three genomes (A, B and C), the string '003' produces a file in which reads do not map to genomes A or B, but map (once or more) to genome C.  The string '--1' would generate a file in which reads uniquely map to genome C. Whether reads  map to genome A or B would be ignored.


for i in *qtrim.gz; do fastq_screen --aligner bowtie2 --conf fastqscreen.conf --subset 0 --tag --filter 0000330 --outdir ./fastq_screen/ $i; done
  
# get back the hit_no_genome reads
fastq_screen --conf ../fastqscreen.conf --nohits 012-X04-P68-COI-ICT_S46_L001_R1_001.fastq.headers.tagged.fastq
no_hits=012-X04-P68-COI-ICT_S46_L001_R1_001.fastq.headers.tagged.tagged_filter.fastq


```

