# Merge assemblies from several species
## Trying de novo mode

-l <label>       name prefix for output transcripts (default: MSTRG)

```bash
#!/bin/bash
#SBATCH --job-name=stringtie
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20

CPU=$SLURM_NPROCS

mkdir -p DENOVO_MODE

for i in $(ls *.sorted.bam)
do
withpath="${i}"
filename=${withpath##*/}
bs="${filename%*.sorted.bam}"
stringtie --rf -p $CPU -l $bs -o DENOVO_MODE/${bs}_transcripts.gtf $i
done

exit
```

Then merge
```bash
cd DENOVO_MODE

ls -1 *_transcripts.gtf > stringtie_gtf_list.txt

stringtie --rf --merge -p 24 -o transcripts.gtf stringtie_gtf_list.txt

# 3) generate FASTA
# Using concat from both genomes, red and green
RNA_REF_FASTA=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/Hybryd/RF_Ref.fna


gffread -w transcripts.fa -g $RNA_REF_FASTA transcripts.gtf

grep "^>" -c transcripts.fa # 76, 658 scaffolds from ctrl samples

```

## run stringtie in reference-guided mode
A reference annotation file in GTF or GFF3 format can be provided to StringTie using the -G option which can be used as 'guides' for the assembly process and help improve the transcript structure recovery for those transcripts.

```bash
#!/bin/bash
#SBATCH --job-name=stringRef
#SBATCH -N 1
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20

mkdir -p REFBASED_MODE;   

RNA_REF_GTF=$1

for i in $(ls *.sorted.bam)
do
withpath="${i}"
filename=${withpath##*/}
bs="${filename%*.sorted.bam}"
stringtie --rf -p 12 -G $RNA_REF_GTF -l $bs -o REFBASED_MODE/${bs}_transcripts.gtf $i
done

exit

```

Run by specie
```bash
# red
sbatch stringtie.sh Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.59.gtf 

# Green
sbatch stringtie.sh FulgensAnnotation.gff3 

# Hybrid
### convert gff to gtf
gffread FulgensAnnotation.gff -T -o FulgensAnnotation.gtf

cat Haliotis_rufescens_gca023055435v1rs.xgHalRufe1.0.p.59.gtf  FulgensAnnotation.gtf > hybrid.gtf

sbatch stringtie_ref.sh hybrid.gtf

# 
cd REFBASED_MODE

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/Red/HISAT2_SAM_BAM_FILES/REFBASED_MODE/* .

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/Green/HISAT2_SAM_BAM_FILES/REFBASED_MODE/* .

ln -s /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/Hybryd/HISAT2_SAM_BAM_FILES/REFBASED_MODE/* .

ls -1 *_transcripts.gtf > stringtie_gtf_list.txt

stringtie --rf --merge -p 24 -o transcripts.gtf stringtie_gtf_list.txt

RNA_REF_FASTA=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/Hybryd/RF_Ref.fna


gffread -w transcripts.fa -g $RNA_REF_FASTA transcripts.gtf

grep "^>" -c transcripts.fa #74226 in comparison w/ 76658 scaffolds from denovo ctrl samples


```

## Assess quality
```bash
sbatch BUSCO.sh transcripts.fa 

```
