# Comparative genomics

In comparative genomics one can:

+ Compare the genome sequence of different species to study what distinguishes different life forms from each other at the molecular level.
+ Study evolutionary changes among organisms
+ Identify conserved or common genes among Species
+ Study the genes that are responsible for specific and unique characteristics for each organism

(Bioinformatic Workbook)


https://gitlab.com/paulklemm_PHD/proteinortho

https://www.vandepeerlab.org/?q=tools/i-adhore30

https://bioinformaticsworkbook.org/dataAnalysis/ComparativeGenomics/OrthofinderSynteny_Update.html#gsc.tab=0

https://github.com/RJEGR/IIIBSS/blob/11ecb90cb37c73110dea393162f3d6f0f94c5ad2/Final_Project/orthoteam.md

# https://gitlab.com/paulklemm_PHD/proteinortho#examples

# using primary isoform
# search how to
The assembly files would contain many transcripts per gene. If we ran OrthoFinder on these raw files it would take 10x longer than necessary and could lower the accuracy. Iâ€™ll use a script provided with OrthoFinder to extract just the longest transcript variant per gene and run OrthoFinder on these files:


my $gene_id;
        
        # if ($acc =~ /^(.*comp\d+_c\d+)_seq/) {
        if ($acc =~ /^(.*comp\d+_c\d+)_seq/) {
            # TRINITY_DN270290_c0
            $gene_id = $1;
        }
        elsif ($acc =~ /^(.*c\d+_g\d+)_i/) {
            # TRINITY_DN270290_c0_g1
            $gene_id = $1;
        }

>TRINITY_DN270290_c0_g1_i1 len=391 path=[0:0-390]

MSTRG.10.1.p1
>MSTRG.10.1.p1
>MSTRG.10.1.p3

# /LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Haliotis/05.Meta_assembly/Protein_orth

```bash
#!/bin/bash
#SBATCH --job-name=ProtOorth
#SBATCH -N 2
#SBATCH --mem=100GB
#SBATCH --ntasks-per-node=20
#SBATCH --mail-type=BEGIN,END
#SBATCH --mail-user=rgomez@uabc.edu.mx

EXPORT=/LUSTRE/bioinformatica_data/genomica_funcional/rgomez/Software/proteinortho-master
export PATH=$PATH:$EXPORT

EXPORT=/LUSTRE/apps/bioinformatica/diamond_v2.1.8/
export PATH=$PATH:$EXPORT

NPROCS=$SLURM_NPROCS
NCPU=$SLURM_NNODES

proteinortho6.pl -threads_per_process=$NPROCS -cpus=$NCPU -project="cds" -p=diamond -step=0 -singles *.cds

proteinortho6.pl -threads_per_process=$NPROCS -cpus=$NCPU -project="pep" -p=diamond -step=0 -singles *.pep

# https://github.com/isabelschober/proteinortho_downstream

python proteinortho_downstream-master/proteinortho_stats.py -p pep.proteinortho.tsv -m all
python proteinortho_downstream-master/proteinortho_stats.py -p cds.proteinortho.tsv -m all

exit

#  (Mandatory) sanitize peptides w/

sed -i -E '/^>/! s/[^XOUBZACDEFGHIKLMNPQRSTVWYxoubzacdefghiklmnpqrstvwy]//g; /^$/d' FF_cross.pep
sed -i -E '/^>/! s/[^XOUBZACDEFGHIKLMNPQRSTVWYxoubzacdefghiklmnpqrstvwy]//g; /^$/d' RR_cross.pep
sed -i -E '/^>/! s/[^XOUBZACDEFGHIKLMNPQRSTVWYxoubzacdefghiklmnpqrstvwy]//g; /^$/d' RF_cross.pep



```